File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs)
    {
        std::unordered_map<std::string, std::vector<std::string> > mp;
        std::vector<std::vector<std::string> > vec;
    
        for (auto s: strs)
        {
            std::string tmp = s;
            std::sort(tmp.begin(), tmp.end());
            mp[tmp].push_back(s);
        }
        for (auto m: mp)
            vec.push_back(m.second);
        return vec;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        res = defaultdict(list)
        
        for s in strs:
            hash_map = [0] * 26
            
            for c in s:
                hash_map[ord(c) - ord('a')] += 1
            
            res[tuple(hash_map)].append(s)
        return res.values()
        